File: ExcelDivider/src/main/java/ru/tsystems/divider/service/impl/NatureServiceImpl.java
package ru.tsystems.divider.service.impl;

import org.springframework.beans.factory.annotation.Autowired;
import ru.tsystems.divider.dao.api.NatureDao;
import ru.tsystems.divider.entity.Nature;
import ru.tsystems.divider.service.api.NatureService;

public class NatureServiceImpl implements NatureService {

    private NatureDao natureDao;

    public NatureServiceImpl( @Autowired NatureDao natureDao) {
        this.natureDao = natureDao;
    }

    public Nature getOrAddNatureByTitle(String title){

        Nature nature = natureDao.getByTitle(title);
        if(nature == null){
            nature = new Nature(title);
            natureDao.persist(nature);
        }
        return nature;
    }

}

File: ExcelDivider/src/main/kt/ru/tsystems/divider/service/impl/NatureServiceImpl.kt
package ru.tsystems.divider.service.impl

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import ru.tsystems.divider.dao.api.NatureDao
import ru.tsystems.divider.entity.Nature
import ru.tsystems.divider.service.api.NatureService

@Service
class NatureServiceImpl(@Autowired private val natureDao: NatureDao) : NatureService {

    override fun getOrAddNatureByTitle(title: String): Nature {

        var nature: Nature? = natureDao.getByTitle(title)
        if (nature == null) {
            nature = Nature(title)
            natureDao.persist(nature)
        }
        return nature
    }

}